### DISCUSSION TITLE
capturing console output

### DISCUSSION THREAD

**Discussion by wjh-97** [Textualize/textual#4004] [Q&A] [answered]
Hey guys! I would greatly appreciate a simple example of how to utilize the "begin_capture_print" function to catch stdout/stderr. My plan is to send console output to a RichLog widget but can't get it to work.

**Comment by davep** [comment, marked as answer]
Often, if you can't get something to work, it can be a good idea to show what you tried and say how it failed so we can help correct any misunderstanding.

To start with, here's an example of an application that logs the time every second, capturing a `print` to do it:

```python
from datetime import datetime

from textual.app import App, ComposeResult
from textual.events import Print
from textual.widgets import Log

class PrintLog(Log):

    def on_mount(self) -> None:
        self.begin_capture_print()

    def on_print(self, event: Print) -> None:
        self.write(event.text)

class PrintLogApp(App[None]):

    def compose(self) -> ComposeResult:
        yield PrintLog()

    def print_time(self) -> None:
        print(str(datetime.now()))

    def on_mount(self) -> None:
        self.set_interval(1.0, self.print_time)

if __name__ == "__main__":
    PrintLogApp().run()
```

**Reply by wjh-97** [reply]
The "on_print" function is what I was missing. Thanks Dave!

**Reply by davep** [reply]
Glad to help. To tie it together: in [the documentation for `begin_capture_print`](https://textual.textualize.io/api/widget/#textual.widget.Widget.begin_capture_print) you'll notice it says:

> If printing is captured, the widget will be sent an [events.Print](https://textual.textualize.io/api/events/#textual.events.Print) message.

So what I added there is [an event handler for the `Print` message](https://textual.textualize.io/guide/events/#message-handlers).
